cdci_data_analysis/analysis/parameters.py

```diff
@@ -397,77 +401,38 @@ def matches_owl_uri(cls, owl_uri: str) -> bool:
     @classmethod
     def from_owl_uri(cls, 
                      owl_uri, 
-                     value=None,
-                     units=None,
-                     name = None,
-                     allowed_units=None,
-                     default_units=None,
-                     units_name=None,
- 
-                     par_format=None,
-                     par_default_format=None,
-                     par_format_name=None,
- 
-                     default_type=None,
-                     allowed_types=None,
- 
-                     check_value=None,
-                     allowed_values=None):
+                     **kwargs):
         # TODO: what about units?
 
         parameter = None
-        
+
         for x in subclasses_recursive(cls):
             logger.debug("searching for class with owl_uri=%s, found %s", owl_uri, x)
             if x.matches_owl_uri(owl_uri):
                 logger.info("will construct %s by url %s", x, owl_uri)
+                call_kwargs = kwargs.copy()
+                call_signature = signature(x)
+                for par_name in kwargs.keys():
+                    if par_name not in call_signature.parameters:
+                        logger.warning("parameter %s with value %s not used to construct %s", par_name, kwargs[par_name], x)
```

Comment:
can we trigger this in the test? when is it even expected to be triggered?

Reply:
Actually, it is not triggered "in the wild". Previously the wrong parameters used to be passed. I added test to trigger it. 

But probably, it is not needed and the exception should be raised...

Reply:
Yeah it's a bit uncertain. Oh, could we try to parse some real output? Store it in test data and parse it? As an additional test.

Reply:
You mean what output? The tests in nb2workflow plugin use real output. But this is irrelevant here, as we need to test only one method `from_owl_url` 
Previously, the exception was caused by bad code, it will not be replicated currently with any data.

Accepting unneeded parameters could potentially simplify future code. On the other hand, it could be more error-prone.

Reply:
perhaps I would extend this warning message also mentioning that the parameter `par_name` won't be used for the instantiation of the object

Reply:
yeah so let's continue there https://github.com/oda-hub/dispatcher-plugin-nb2workflow/issues/23#issuecomment-1241649054

Reply:
> > perhaps I would extend this warning message also mentioning that the parameter `par_name` won't be used for the instantiation of the object
> 
> But it will be.

I am lost what on we are talking about. `call_kwargs` is used to construct, and you pop from it, right? And there is a warning?

Reply:
Sorry, I've read wrongly, confused myself, then confused everyone. I've removed this comment

Reply:
my suggestion was to extend the log message mentioning that the object will be built without using `par_name` since this is not part of the signature, I also was not clear enough probably

Reply:
> my suggestion was to extend the log message mentioning that the object will be built without using `par_name` since this is not part of the signature, I also was not clear enough probably

Could you propose the exact phrasing?

Reply:
`parameter %s, with value %s, is not used to construct a %s object, therefore this will be discarded for the instantiation`